tox.ini

```diff
@@ -92,6 +92,41 @@ skip_install = true
 allowlist_externals =
   git
 
+
+[testenv:preview]
+description = preview the docs
+deps =
+  sphinx-autobuild
+  -r{toxinidir}/docs/requirements.txt
+  # FIXME: re-enable the "-r" + "-c" paradigm once the pip bug is fixed.
+  # Ref: https://github.com/pypa/pip/issues/9243
+  # -r{toxinidir}/docs/requirements.in
+  # -c{toxinidir}/docs/requirements.txt
+commands_pre =
+commands =
+  # Retrieve possibly missing commits:
+  -git fetch --unshallow
+  -git fetch --tags
+
+  # Build the html docs with sphinx-autobuild:
+  sphinx-autobuild \
+    -j auto \
+    -b html \
+    -n \
+    -W \
+    . \
+    --watch .. \
```

Comment:
I had to watch the whole root directory of the project because it contains the CHANGELOG and README. 

Can I create symlinks of those files into `docs/`? Then we can remove this line, which would probably be better.

Reply:
No, you don't need that. @pradyunsg shown me recently that you can add an extra CLI option to watch more files. Just need to google that and make use of it...

Reply:
Can you share your wisdom?

Reply:
*/me just googles that*

Reply:
There's nothing clear in the README, but the CLI parser spec says `--watch` is incremental, so we could probably just use `--watch docs/ --watch CHANGELOG.md --watch README.rst`. Though, it's unclear if it would stomach a file path in place of a directory path..

Reply:
I think that should work. I checked that https://github.com/executablebooks/sphinx-autobuild/blob/ea4d04f/src/sphinx_autobuild/cli.py#L187-L191 calls `Server.watch()` on all the corresponding CLI options passed. And https://github.com/lepture/python-livereload/blob/d042fd3/livereload/server.py#L225C26-L226 implies that the values could be file/dir paths or even path wildcards.

`sphinx-autobuild` other possible values which could be a chance to contribute it.

Reply:
> There's nothing clear in the README, but the CLI parser spec says `--watch` is incremental, so we could probably just use `--watch docs/ --watch CHANGELOG.md --watch README.rst`. Though, it's unclear if it would stomach a file path in place of a directory path..

That works! Hm, I could have sworn I tested that and received some traceback about the given path not being a directory.

Reply:
Maybe you tried that under a different chdir?